home/models/survey.py

```diff
@@ -0,0 +1,151 @@
+
+    def save(
+        self, force_insert=False, force_update=False, using=None, update_fields=None
+    ):
+        self.slug = slugify(self.name)
+        if update_fields is not None and "name" in update_fields:
+            update_fields = {"slug"}.union(update_fields)
+        super().save(
+            force_insert=force_insert,
+            force_update=force_update,
+            using=using,
+            update_fields=update_fields,
+        )
+
+
+class TypeField(models.TextChoices):
+    TEXT = "TEXT", _("Text")
+    NUMBER = "NUMBER", _("Number")
+    DATE = "DATE", _("Date")
+    RADIO = "RADIO", _("Radio")
+    SELECT = "SELECT", _("Select")
+    MULTI_SELECT = "MULTI_SELECT", _("Multi Select")
+    TEXT_AREA = "TEXT_AREA", _("Text Aria")
+    URL = "URL", _("URL")
+    EMAIL = "EMAIL", _("Email")
+    RATING = "RATING", _("Rating")
+
+
+class Question(BaseModel):
+    key = models.CharField(
+        max_length=225,
+        unique=True,
+        blank=True,
+        help_text=_(
+            "Unique key for this question, fill in the blank if you want to use for automatic generation."
+        ),
+    )
+    survey = models.ForeignKey(
+        Survey, related_name="questions", on_delete=models.CASCADE
+    )
+    label = models.CharField(
+        max_length=500, help_text=_("Enter your question in here.")
+    )
+    type_field = models.CharField(choices=TypeField.choices, max_length=100)
+    choices = models.TextField(
+        blank=True,
+        help_text=_(
+            "If type of field is radio, select, or multi select, fill in the options separated by commas. Ex: Male, Female."
+        ),
+    )
+    help_text = models.CharField(
+        max_length=200,
+        blank=True,
+        help_text=_("You can add a help text in here."),
+    )
+    required = models.BooleanField(
+        default=True,
+        help_text=_("If True, the user must provide an answer to this question."),
+    )
+    ordering = models.PositiveIntegerField(
+        default=0, help_text=_("Defines the question order within the surveys.")
+    )
+
+    class Meta:
+        ordering = ["ordering"]
+
+    def __str__(self):
+        return f"{self.label}-survey-{self.survey.id}"
+
+    def save(
+        self, force_insert=False, force_update=False, using=None, update_fields=None
+    ):
+        self.slug = slugify(self.label)
+        if update_fields is not None and "label" in update_fields:
+            update_fields = {"slug"}.union(update_fields)
+        super().save(
+            force_insert=force_insert,
+            force_update=force_update,
+            using=using,
+            update_fields=update_fields,
+        )
+
+
+class UserAnswer(BaseModel):
+    survey = models.ForeignKey(Survey, on_delete=models.CASCADE)
+    user = models.ForeignKey("accounts.CustomUser", on_delete=models.CASCADE)
+
+    class Meta:
+        ordering = ["-updated_at"]
+
+    def __str__(self):
+        return str(self.id)
+
+
+class Answer(BaseModel):
```

Comment:
And then this could be called `UserAnswer` (I could see us eventually creating a generic `Answer` model).